contracts/socket/SocketDst.sol

```diff
@@ -200,34 +192,32 @@ abstract contract SocketDst is SocketBase {
      * code exists in the given address.
      */
     function _execute(
-        address executor,
-        uint256 executionFee,
+        address executor_,
         address localPlug_,
         uint32 remoteChainSlug_,
-        uint256 msgGasLimit_,
-        bytes32 msgId_,
-        bytes calldata payload_
+        ISocket.MessageDetails memory messageDetails_
     ) internal {
+        if (msg.value < messageDetails_.msgValue) revert InsufficientMsgValue();
```

Comment:
When we change to decoding msgValue from the extraParam, socket will not do this check. The check needs to be done on execution manager where decoding logic is present.